### PR TITLE
Initial psycopg plugin

### PR DESCRIPTION
**Pull request by tom-pytel**
I will do the other plugin stuff once the test is stable.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@tom-pytel Take a look at slack. I left a message for your Apache committer to further process. It is in your hands to complete them.

**Comment by tom-pytel** [comment]
Hey @kezhenxu94, I have no idea why the test is failing. I copied the framework from pymsql test which works and the code in `provider.py` works by itself but for some reason running as a test it is getting:
```
Exception: Wait time exceeded 100 sec. Exception HTTPConnectionPool(host='0.0.0.0', port=9090): Max retries exceeded with url: /users (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc586e6b040>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

File: tests/plugin/sw_psycopg2/docker-compose.yml
Comment (kezhenxu94) [inline tests/plugin/sw_psycopg2/docker-compose.yml:36]: Hi, this is the reason why tests failed, the exposed port is 5432 but this checks 3306, I believe it's a copy-paste typo.

Screen Shot 2021-06-06 at 09 42 31

```suggestion
      test: ["CMD", "bash", "-c", "cat < /dev/null > /dev/tcp/127.0.0.1/5432"]
```

**Comment by tom-pytel** [comment]
Ok, one thing to resolve before merge. I reused `config.mysql_trace_sql_parameters` and `config.mysql_sql_parameters_max_length` options in this plugin just to get it running but the final param should be one of two cases:
1. Add separate `config.postgre_trace_sql_parameters` and `config.postgre_sql_parameters_max_length`.
2. Convert `config.mysql_trace_sql_parameters` to just `config.trace_sql_parameters` and have all SQL agents share this to avoid explosion of options as different SQL agents are added (since odds are you will not be using mysql and postgresql simultaneously in the same project) - this is my preferred option.

And another idea unrelated to which option above is taken. If you are not locked into the current options scheme - reduce the two parameters `config.mysql_trace_sql_parameters` and `config.mysql_sql_parameters_max_length` into a single option `config.mysql_parameters` (or something like that) which gives the length and if length 0 is specified (default) then disables parameters?

**Comment by kezhenxu94** [comment]
> And another idea unrelated to which option above is taken. If you are not locked into the current options scheme - reduce the two parameters `config.mysql_trace_sql_parameters` and `config.mysql_sql_parameters_max_length` into a single option `config.mysql_parameters` (or something like that) which gives the length and if length 0 is specified (default) then disables parameters?

This is an optimization I've been thinking about, I'm +1 to it.

Also, I prefer (2.), since there is rare situation that a service will use multiple DB layers and not use the same configurations for them.

File: skywalking/__init__.py
Comment (kezhenxu94) [inline skywalking/__init__.py:43]: Remember to add this in main repo and UI repo

**Review by kezhenxu94 — COMMENTED** [review]
Please also add the plugin to https://github.com/apache/skywalking-python/blob/master/docs/Plugins.md 🙇

File: skywalking/__init__.py
Comment (tom-pytel) [reply skywalking/__init__.py:43]: I've got that done just was waiting for this part to finalize.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**
